cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -37,14 +37,22 @@ func Run(gadgetRoot, device string, options *Options) error {
 		return fmt.Errorf("cannot use empty device node")
 	}
 
-	// XXX: ensure we test that the current partition table is
-	//      compatible with the gadget
 	lv, err := gadget.PositionedVolumeFromGadget(gadgetRoot)
 	if err != nil {
 		return fmt.Errorf("cannot layout the volume: %v", err)
 	}
 
 	sfdisk := partition.NewSFDisk(device)
+	diskLayout, err := sfdisk.Layout()
```

Comment:
Looking again at `sfdisk.Layout()`, perhaps it's time to introduce something like:
```
package sfdisk
...
type DeviceLayout struct {
    Structure []DeviceStructure  // sorted by the start offset
    Id string // disk ID
    Device string // disk device, eg. /dev/sda
}
```
to make it clear that it's not exactly the same as gadget.Volume

Reply:
I created `DeviceLayout` to hold layout information and it solved an API design issue related to `Layout()` being called to set up the internal state to be used by `Create()`. `Create()` can now be a method of `DeviceLayout` and all the necessary setup is made in `NewDeviceLayout()`.